### PR TITLE
Tournaments: support custom format names

### PR DESCRIPTION
**Pull request by sirDonovan**

### CODE REVIEW

File: js/client-chat-tournament.js [hunk 1]
```diff
@@ -355,10 +355,16 @@
 						this.info.isActive = true;
 					}
 
-					if ('format' in this.updates || 'isJoined' in this.updates) {
+					if ('format' in this.updates) {
 						this.$format.text(window.BattleFormats && BattleFormats[this.info.format] ? BattleFormats[this.info.format].name : this.info.format);
 						this.updateTeams();
 					}
+					if ('customFormat' in this.updates) {
```
Comment (Zarel) [inline js/client-chat-tournament.js]: Why not just send the custom format name in the `format` field? You seem to be handling them roughly the same way.

File: js/client-chat-tournament.js [hunk 2]
```diff
@@ -355,10 +355,16 @@
...
+						this.$format.text(window.BattleFormats && BattleFormats[this.info.customFormat] ? BattleFormats[this.info.customFormat].name : this.info.customFormat);
+					}
+					if ('isJoined' in this.updates) {
```
Comment (Zarel) [inline js/client-chat-tournament.js]: Is this related?

File: js/client-chat-tournament.js (hunk 1 above)
Comment (sirDonovan) [inline js/client-chat-tournament.js]: The <code>format</code> field is also used for the team selector.

File: js/client-chat-tournament.js (hunk 2 above)
Comment (sirDonovan) [inline js/client-chat-tournament.js]: Yes, it prevents joining the tournament from resetting the custom format name.

File: js/client-chat-tournament.js (hunk 1 above)
Comment (Zarel) [inline js/client-chat-tournament.js]: `teambuilderFormat` may be a better name

File: js/client-chat-tournament.js (hunk 2 above)
Comment (Zarel) [inline js/client-chat-tournament.js]: How?

Comment (sirDonovan) [inline js/client-chat-tournament.js]: <code>'isJoined' in this.updates</code> was in the same if statement as <code>'format' in this.updates</code>, so the format name was set whenever the user joined the tournament.

Comment (Zarel) [inline js/client-chat-tournament.js]: Oh, I understand now, thanks.

File: js/client-chat-tournament.js (hunk 1 above)
Comment (sirDonovan) [inline js/client-chat-tournament.js]: Maybe, but making it backwards compatible gets a bit messy.

Comment (Zarel) [inline js/client-chat-tournament.js]: Why? `teambuilderFormat` can inherit from `format` if it doesn't exist.

Comment (sirDonovan) [inline js/client-chat-tournament.js]: I'm not sure how I was thinking of it before, but yeah that's fine.

File: js/client-chat-tournament.js [hunk 3]
```diff
@@ -355,8 +356,15 @@
...
 					}
 
-					if ('format' in this.updates || 'isJoined' in this.updates) {
-						this.$format.text(window.BattleFormats && BattleFormats[this.info.format] ? BattleFormats[this.info.format].name : this.info.format);
+					if ('format' in this.updates || 'teambuilderFormat' in this.updates) {
+						if (!this.info.teambuilderFormat) this.info.teambuilderFormat = this.info.format;
+						this.$format.text(window.BattleFormats && BattleFormats[this.info.teambuilderFormat] ? BattleFormats[this.info.teambuilderFormat].name : this.info.teambuilderFormat);
+						this.updateTeams();
+					}
+					if ('customFormat' in this.updates) {
```
Comment (Zarel) [inline js/client-chat-tournament.js]: ?

Is this still important for something?

Comment (sirDonovan) [inline js/client-chat-tournament.js]: Whoops, no I removed that.

**Comment by Zarel** [comment]
Looks good, but I'll keep this open for a bit for review...

**Comment by sirDonovan** [comment]
Any further comments on this?

**Comment by Zarel** [comment]
I don't understand it, but it doesn't look like a big maintenance burden.
